# Bioinformatics

Reviews on different NGS data analysis methods. 
LaTeX Beamer format.

### Content
* [atacseq](https://github.com/olegs/bioinformatics/blob/master/atacseq/pipeline.pdf) - summary on ATAC-Seq processing methods
* [chipseq](https://github.com/olegs/bioinformatics/blob/master/chipseq/chipseq.pdf) - summary on ChIP-Seq processing from qc to replicated peak calling
* [differential_chipseq](https://github.com/olegs/bioinformatics/blob/master/differential_chipseq/diffchipseq.pdf) - how to compare replicated data
* [peaks](https://github.com/olegs/bioinformatics/blob/master/peaks/peaks.pdf) - notes on peaks classification by length based on small subset of [Roadmap Epigenomics](http://www.roadmapepigenomics.org/) data 

### ENCODE resources
ENCODE standards <https://www.encodeproject.org/data-standards>  
ENCODE Rest API <https://www.encodeproject.org/help/rest-api>  
ENCODE DNANexus pipelines source <https://github.com/ENCODE-DCC>  

### Others
ChIP-seq analysis notes from Tommy Tang <https://github.com/crazyhottommy/ChIP-seq-analysis>
